### PR TITLE
Herwig lhe matching fix 10_6

### PR DESCRIPTION
**Pull request by Dominic-Stafford**
Backport of https://github.com/cms-sw/cmssw/pull/42673 for use in Run 2 UL gen production. These add a new Herwig7HadronizerFilter, designed to resolve the issue of the wrong lhe events being saved with Herwig.

Should be tested with https://github.com/cms-sw/cmsdist/pull/8728

### CODE REVIEW

**Comment by antoniovilela** [comment]
@Dominic-Stafford @cms-sw/generators-l2 
Hello, has this been validated in 10_6? Will we need a new 10_6 release?

**Comment by antoniovilela** [comment]
please test with https://github.com/cms-sw/cmsdist/pull/8728

**Comment by antoniovilela** [comment]
backport of #42673

**Comment by menglu21** [comment]
> @Dominic-Stafford @cms-sw/generators-l2 Hello, has this been validated in 10_6? Will we need a new 10_6 release?

Hi, I don't think it's urgent since herwig is not used that much. This can be done in the future when some necessary update needed for 10_6

**Comment by Dominic-Stafford** [comment]
> @Dominic-Stafford @cms-sw/generators-l2 Hello, has this been validated in 10_6? Will we need a new 10_6 release?

Hi, yes, this has been validated in 10_6. It would be nice to have in reasonably soon since it's a reasonably large change to Herwig, but as Meng says Herwig isn't used for so many requests, so it probably doesn't need its own release

File: SimDataFormats/GeneratorProducts/src/classes_def.xml [hunk 1]
```diff
@@ -206,11 +206,12 @@
 	<class name="LHERunInfoProduct" ClassVersion="10">
   <version ClassVersion="10" checksum="2098560362"/>
  </class>
-	<class name="LHEEventProduct" ClassVersion="13">
+	<class name="LHEEventProduct" ClassVersion="14">
   <version ClassVersion="10" checksum="1026978494"/>
   <version ClassVersion="11" checksum="619154414"/>
   <version ClassVersion="12" checksum="259352862"/>
   <version ClassVersion="13" checksum="3927731647"/>
+  <version ClassVersion="14" checksum="395176982"/>
```
Comment (makortel) [inline SimDataFormats/GeneratorProducts/src/classes_def.xml:214]: In retrospect the https://github.com/cms-sw/cmssw/pull/25239 should have been backported either along this PR or before. Now files produced with 10_6_38 or later contain a version of `LHEEventProduct` that never existed in the `master` branch, and make reading such files in later releases more difficult. See https://github.com/cms-sw/cmssw/issues/49538#issuecomment-3849339297 for more details.

**Pull request merged**